src/BondEvaluatorDoubleWell.h

```diff
@@ -2,151 +2,132 @@
-\frac{c}{2b}\left(r-a/2\right)+c/2
-
-\f}
-which has two minima at r = (a/2 +/- b), seperated by a maximum at a/2 of height V_max when c is set
-to zero.
-
-The parameter a tunes the location of the maximal value and the parameter b tunes the distance of
-the two maxima from each other.  This potential is useful to model bonds which can be either
-mechanically or thermally "activated" into a effectively longer state. The value of V_max can be
-used to tune the height of the energy barrier in between the two states.
-
-If c is non zero, the relative energy of the minima can be tuned, where c is the energy of the
-second minima, the first minima value is at zero. This  causes a small shift in the location of the
-minima and the maxima, because of the added linear term.
-
-The parameters are:
-    - \a V_max (params.x) potential difference between the the first minima and maxima
-    - \a a (params.y) shift for the location of the V_max, the maximum is at approx. a/2
-    - \a b (params.z) scaling for the distance of the two minima at approx. (a/2 +/- b)
-    - \a c (params.w) potential difference between the two minima, default is c=0
-
-*/
-class BondEvaluatorDoubleWell
+//! Parameters of double well bond potential
+struct BondParametersDoubleWell : public BondParameters
     {
-    public:
-    //! Define the parameter type used by this bond potential evaluator
-
-    typedef Scalar4 param_type;
+#ifndef __HIPCC__
+    BondParametersDoubleWell() : r_1(0), r_diff(1.0), U_1(0), U_tilt(0) { }
 
-    //! Constructs the pair potential evaluator
-    /*!
-     * \param _rsq Squared distance beteen the particles
-     * \param _params Per type bond parameters of this potential as given above
-     */
-    DEVICE BondEvaluatorDoubleWell(Scalar _rsq, const param_type& _params)
-        : rsq(_rsq), V_max(_params.x), a(_params.y), b(_params.z), c(_params.w)
+    BondParametersDoubleWell(pybind11::dict v)
         {
+        r_1 = v["r_1"].cast<Scalar>();
+        r_diff = r_1 - v["r_0"].cast<Scalar>();
+        U_1 = v["U_1"].cast<Scalar>();
+        U_tilt = v["U_tilt"].cast<Scalar>();
         }
 
-    //! This evaluator doesn't use diameter information
-    DEVICE static bool needsDiameter()
+    pybind11::dict asDict()
         {
-        return false;
+        pybind11::dict v;
+        v["r_0"] = r_1 - r_diff;
+        v["r_1"] = r_1;
+        v["U_1"] = U_1;
+        v["U_tilt"] = U_tilt;
+        return v;
         }
+#endif
+
+    Scalar r_1; //!<
+    Scalar r_diff;
+    Scalar U_1;    //!<
+    Scalar U_tilt; //!<
+    }
+#if HOOMD_LONGREAL_SIZE == 32
+    __attribute__((aligned(16)));
+#else
+    __attribute__((aligned(32)));
+#endif
 
-    //! Accept the optional diameter values
-    /*!
-     * \param da Diameter of particle a
-     * \param db Diameter of particle b
-     */
-    DEVICE void setDiameter(Scalar da, Scalar db) { }
+//! Class for evaluating the double well bond potential
```

Comment:
This documentation needs to be updated  - it can be left for later, but then we need to leave a TODO note at least